front/lib/auth.ts

```diff
@@ -297,3 +300,59 @@ export async function getAPIKey(
 
   return new Ok(key);
 }
+
+const DEFAULT_DATASOURCES_COUNT_LIMIT = 1;
+const DEFAULT_DATASOURCES_DOCUMENTS_COUNT_LIMIT = 32;
+const DEFAULT_DATASOURCES_DOCUMENTS_SIZE_MB_LIMIT = 1;
+
+function planForWorkspace(w: Workspace): PlanType {
+  let limits = {
+    dataSources: {
+      count: DEFAULT_DATASOURCES_COUNT_LIMIT,
+      documents: {
+        count: DEFAULT_DATASOURCES_DOCUMENTS_COUNT_LIMIT,
+        sizeMb: DEFAULT_DATASOURCES_DOCUMENTS_SIZE_MB_LIMIT,
+      },
+    },
+  };
+
+  if (w.plan) {
```

Comment:
Can we use a little parse_payload here?
ChatGPT generated that:
```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "LimitsType": {
      "type": "object",
      "properties": {
        "dataSources": {
          "type": "object",
          "properties": {
            "count": { "type": "number" },
            "documents": {
              "type": "object",
              "properties": {
                "count": { "type": "number" },
                "sizeMb": { "type": "number" }
              },
              "required": ["count", "sizeMb"],
              "additionalProperties": false
            }
          },
          "required": ["count", "documents"],
          "additionalProperties": false
        }
      },
      "required": ["dataSources"],
      "additionalProperties": false
    },
    "PlanType": {
      "type": "object",
      "properties": {
        "limits": { "$ref": "#/definitions/LimitsType" }
      },
      "required": ["limits"],
      "additionalProperties": false
    }
  }
}

```

I did not test it though.

Here is the prompt:
```
Can you generate the AJV JSONSchemaType for the following interface?
export type LimitsType = {
  dataSources: { count: number; documents: { count: number; sizeMb: number } };
};

export type PlanType = {
  limits: LimitsType;
};```